File: BookLibrary-cdi/frontend/src/app/model/book-filter.ts
export class BookFilter {
    name: string;
    authorName: string;
    ratingFrom: number;
    ratingTo: number;
    year: number;
    isbn: string;

    constructor(name: string,
                authorName: string,
                bookRatingFrom: number,
                bookRatingTo: number,
                year: number,
                isbn: string) {
        this.name = name;
        this.authorName = authorName;
        this.ratingFrom = bookRatingFrom;
        this.ratingTo = bookRatingTo;
        this.year = year;
        this.isbn = isbn;
    }
}

File: BookLibrary-cdi/backend/src/main/java/com/softserveinc/booklibrary/backend/dto/filtering/BookFilter.java
package com.softserveinc.booklibrary.backend.dto.filtering;

import java.math.BigDecimal;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class BookFilter {

	private String name;
	private AuthorFilter authorFilter;  // todo: why are you object here?
	private BigDecimal bookRatingFrom;  // todo rename to ratingFrom
	private BigDecimal bookRatingTo;    // todo rename to ratingTo
	private Integer year;
	private String isbn;

}
